grunt/shell.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+const ghPagesList = [
+  'index.html',
+  'favicon.ico',
+  'public',
+].join(' ');
+
+module.exports = {
+  'git-is-clean': {
+    command: 'test -z "$(git status --porcelain)"  || (git status && false)',
```

Comment:
Would you mind adding a comment about why the logic is constructed in this way? How does each predicate give us what we want?